src/habitat.cr

```diff
@@ -268,10 +275,13 @@ class Habitat
           @@{{ decl.var }} = value
         end
 
-        def self.{{ decl.var }}
+        def self.{{ decl.var }} : {{ decl.type }}{% if nilable %}?{% end %}
           @@{{ decl.var }}{% if !nilable %}.not_nil!{% end %}
         end
 
+        # Used for checking missing settings on non-nilable types
+        # It's advised to use {{ decl.var }} in your apps to ensure
+        # the propper type is checked.
         def self.{{ decl.var }}?
```

Comment:
We can't really make this method private, but I think people should really just use the other method in their apps. This is essentially an internal method at this point. Do protected class methods exist? 🤔